agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -653,6 +653,70 @@ bool monitor_thread::update_ap_stats()
     return true;
 }
 
+bool monitor_thread::_start_monitoring_sta(const std::string &sta_mac,
+                                           const std::string &set_bridge_4addr_mac,
+                                           const int vap_id)
+{
+    auto sta_node = mon_db.sta_find(sta_mac);
+    if (sta_node) {
+        mon_db.sta_erase(sta_mac);
+    }
+
+    auto vap_node = mon_db.vap_get_by_id(vap_id);
+    if (vap_node == nullptr) {
+        LOG(ERROR) << "vap_id " << vap_id << " doesn't not exists";
+        return false;
+    }
+
+    sta_node = mon_db.sta_add(sta_mac, vap_id);
+    sta_node->set_bridge_4addr_mac(set_bridge_4addr_mac);
+
+#ifdef BEEROCKS_RDKB
+    //clean rdkb monitor data if already in database.
+    auto client = mon_rdkb_hal.conf_get_client(sta_mac);
+    if (client != nullptr) {
+        client->setStartTime(std::chrono::steady_clock::now());
+        client->setLastSampleTime(std::chrono::steady_clock::now());
+        client->setAccumulatedPackets(0);
+        client->clearData();
+    }
+#endif
+    return true;
+}
+
+bool monitor_thread::_monitored_sta_set_ipv4(const std::string &sta_mac,
+                                             const std::string &sta_ipv4)
+{
+    auto sta_node = mon_db.sta_find(sta_mac);
+    if (!sta_node) {
+        LOG(ERROR) << "Cannot find STA with MAC " << sta_mac;
+        return false;
+    }
+
+    sta_node->set_ipv4(sta_ipv4);
+    return true;
+}
+
+bool monitor_thread::start_monitoring_sta(const std::string &sta_mac, const std::string &sta_ipv4,
+                                          const std::string &set_bridge_4addr_mac, const int vap_id)
+{
+    const std::lock_guard<std::mutex> lock(monitoring_sta_mutex);
+
+    if (!_start_monitoring_sta(sta_mac, set_bridge_4addr_mac, vap_id)) {
+        LOG(ERROR) << "_start_monitoring_sta with mac " << sta_mac << ", bridge_mac "
+                   << set_bridge_4addr_mac << ", vap_id " << vap_id << " failed!";
+        return false;
+    }
+
+    if (!_monitored_sta_set_ipv4(sta_mac, sta_ipv4)) {
```

Comment:
Why a separate method for setting the IP address?
Can't you just pass it to `_start_monitoring_sta()`?

Reply:
At the moment, I am moving in direction described by https://github.com/prplfoundation/prplMesh/pull/1139#pullrequestreview-398271772 .
BWL will give me `mac` and `vap_id` at the same time, but `ipv4` will arrive a bit later from the slave.